script.js

```diff
@@ -1,4 +1,294 @@
+    console.log(`player 1 dice 2 array, ${player1dice2Arr}`);
+
+    //Summary Result Message + Note on how to choose dice order
+    myOutputValue = `Your rolled dice result: <br><br>
+  dice 1 : ${player1dice1Arr} <br><br>
+  dice 2 : ${player1dice2Arr} <br><br>
+  Please choose the order of desired dice result. <br>
+  Type 1 and click enter if you wish for dice 1 to be the first number <br>
+  Type 2 and click enter if you wish for dice 2 to be the first number`;
+
+    currentGameMode = "player1Choice"; //Change game mode to allow player 1 to decide dice order
+    console.log(`diceResult of both dice rolled, ${myOutputValue}`);
+  }
+  // Chosen dice 1 to be first
+  else if (currentGameMode == "player1Choice" && userChosenDiceOrder == 1) {
+    myOutputValue = `${player1} have chose dice 1 to be the first in order, your result is ${player1dice1Arr}${player1dice2Arr} <br><br>
+     ${player2}, you may now begin to roll the dice by clicking enter.`;
+
+    player1Score = `${player1dice1Arr}${player1dice2Arr}`; // Player 1 score
+
+    console.log(`player 1 score, ${player1Score}`);
+    console.log(`user Chosen Dice Order, ${userChosenDiceOrder}`);
+    console.log(`Dice 1 as first number results, ${myOutputValue}`);
+
+    currentGameMode = "player2Roll"; //Change game mode to player 2 turn to roll
+    console.log(
+      `Player 1 turn end, now is player 2 turn. Current mode : ${currentGameMode}`
+    );
+
+    return myOutputValue;
+  } else if (currentGameMode == "player1Choice" && userChosenDiceOrder == 2) {
+    myOutputValue = `${player1} have chose dice 2 to be the first in order, your result is ${player1dice2Arr}${player1dice1Arr} <br><br>
+     ${player2}, you may now begin to roll the dice by clicking enter.`;
+
+    player1Score = `${player1dice2Arr}${player1dice1Arr}`; // Player 1 score
+
+    console.log(`player 1 score, ${player1Score}`);
+    console.log(`user Chosen Dice Order, ${userChosenDiceOrder}`);
+    console.log(`Dice 2 as first number results, ${myOutputValue}`);
+
+    currentGameMode = "player2Roll"; // Change game mode to player 2 turn to roll
+    console.log(
+      `Player 1 turn end, now is player 2 turn. Current mode : ${currentGameMode}`
+    );
+  } else if (
+    currentGameMode == "player1Choice" &&
+    (userChosenDiceOrder !== 2 || userChosenDiceOrder !== 1)
+  ) {
+    myOutputValue = `You have enter an invalid input for dice order. Pleaser enter either a 1 or 2 to choose your dice order. Thank you! <br><br>
+    Your rolled dice result: <br><br>
+  dice 1 : ${player1dice1Arr} <br><br>
+  dice 2 : ${player1dice2Arr} <br><br>
+  Please choose the order of desired dice result. <br>
+  Type 1 and click enter if you wish for dice 1 to be the first number <br>
+  Type 2 and click enter if you wish for dice 2 to be the first number
+    `;
+
+    console.log(
+      `Invalid input, current mode should still be player choice : ${currentGameMode}`
+    );
+    return myOutputValue;
```

Comment:
this is fine but i would probably put the negative case up top first and check for that, returning early if so before moving on with the rest of my code

i like that you display the invalid input message to the user along with the dice rolls again! a comment i have for many students is that when they output this msg, they don't show the rolls again and it is very possible the user forgot the dice rolls